shell/browser/api/electron_api_web_frame_main.cc

```diff
@@ -0,0 +1,230 @@
+  return render_frame_->GetParent();
+}
+
+std::vector<content::RenderFrameHost*> WebFrame::Frames(gin::Arguments* args) {
+  std::vector<content::RenderFrameHost*> frame_hosts;
+  CHECK_RENDER_FRAME(frames, args, frame_hosts);
+
+  for (auto* rfh : render_frame_->GetFramesInSubtree()) {
+    if (rfh->GetParent() == render_frame_)
+      frame_hosts.push_back(rfh);
+  }
+
+  return frame_hosts;
+}
+
+std::vector<content::RenderFrameHost*> WebFrame::FramesInSubtree(
+    gin::Arguments* args) {
+  std::vector<content::RenderFrameHost*> frame_hosts;
+  CHECK_RENDER_FRAME(frames, args, frame_hosts);
+
+  for (auto* rfh : render_frame_->GetFramesInSubtree()) {
+    frame_hosts.push_back(rfh);
+  }
+
+  return frame_hosts;
+}
+
+// static
+gin::Handle<WebFrame> WebFrame::From(v8::Isolate* isolate,
+                                     content::RenderFrameHost* rfh) {
+  if (rfh == nullptr)
+    return gin::Handle<WebFrame>();
+  auto frame_map = g_render_frame_map.Get();
+  auto iter = frame_map.find(rfh);
+  WebFrame* web_frame =
+      iter == frame_map.end() ? new WebFrame(rfh) : iter->second;
+  auto handle = gin::CreateHandle(isolate, web_frame);
+  return handle;
+}
+
+// static
+gin::Handle<WebFrame> WebFrame::FromID(v8::Isolate* isolate,
+                                       int render_process_id,
+                                       int render_frame_id) {
+  auto* rfh =
+      content::RenderFrameHost::FromID(render_process_id, render_frame_id);
+  return From(isolate, rfh);
+}
+
+// static
+void WebFrame::RenderFrameDeleted(content::RenderFrameHost* rfh) {
+  auto frame_map = g_render_frame_map.Get();
+  auto iter = frame_map.find(rfh);
+  if (iter == frame_map.end())
+    return;
+  WebFrame* web_frame = iter->second;
+  web_frame->MarkRenderFrameDisposed();
+}
+
+gin::ObjectTemplateBuilder WebFrame::GetObjectTemplateBuilder(
+    v8::Isolate* isolate) {
+  return gin::Wrappable<WebFrame>::GetObjectTemplateBuilder(isolate)
+      .SetMethod("executeJavaScript", &WebFrame::ExecuteJavaScript)
+      .SetMethod("reload", &WebFrame::Reload)
+      .SetProperty("frameTreeNodeId", &WebFrame::FrameTreeNodeID)
+      .SetProperty("processId", &WebFrame::ProcessID)
+      .SetProperty("routingId", &WebFrame::RoutingID)
+      .SetProperty("url", &WebFrame::URL)
+      .SetProperty("top", &WebFrame::Top)
+      .SetProperty("parent", &WebFrame::Parent)
+      .SetProperty("frames", &WebFrame::Frames)
+      .SetProperty("framesInSubtree", &WebFrame::FramesInSubtree);
+}
+
+const char* WebFrame::GetTypeName() {
+  return "WebFrame";
+}
+
+}  // namespace api
+
+}  // namespace electron
+
+namespace {
+
+using electron::api::WebFrame;
+
+v8::Local<v8::Value> FromID(gin_helper::ErrorThrower thrower,
+                            int render_process_id,
+                            int render_frame_id) {
+  if (!electron::Browser::Get()->is_ready()) {
+    thrower.ThrowError("WebFrame can only be received when app is ready");
```

Comment:
nit:
```suggestion
    thrower.ThrowError("WebFrame is available only after app ready");
```

Reply:
I agree with your suggestion, however, this was copied from elsewhere in the codebase. The same message is used for the session and webframe APIs.

If we change it here, I imagine that'd make it harder for folks to find the solution if they searched the error message. Or we could change it for all of them. Or maybe I'm overthinking it. 😅

Reply:
Yeah... this error isn't consistent in our codebase (there are already a few variations). The one I picked here is one of the more grammatically fluent variations. I'm not sure I understand your point about searching the error message, it should be just as searchable as it remains unique.

Reply:
If the error message remains similar to others, finding it in search engines might be easier. Since you mentioned we use the grammatically improved version elsewhere, I'll switch to that. 👍